src/components/NewStudentForm.tsx

```diff
@@ -0,0 +1,104 @@
+import  {useEffect, useState} from 'react';
+import * as React from "react"
+import { Button, Form, FormGroup, Input, Label } from "reactstrap";
+import axios from "axios";
+import { API_URL } from "../constants";
+
+const initialState = {
+    pk: 0,
+    name: "",
+    email: "",
+    document: "",
+    phone: ""
+}
+
+export interface studentInterface {
+    pk: number,
+    name: string,
+    email: string,
+    document: string,
+    phone: string
+}
+
+interface propInterface {
+    resetState: () => void,
+    student?: studentInterface,
+    toggle: () => void
+}
+
+function NewStudentForm(props:propInterface) {
+
+    const [state, setState] = useState<studentInterface>(initialState)
```

Comment:
I'd generally avoid using using `state` and `setState`, and opt for more verbose names. Makes things more readable.

```suggestion
    const [student, setStudent] = useState<studentInterface>(initialState)
```